src/mappings/events/vaults.ts

```diff
@@ -94,8 +94,14 @@ export const vaultsEventKit = (block: any, data: any, module: string, path: stri
       (vaultState as any)[oldProperty] -= BigInt(1);
     }
 
-    if (newState && propertyMap[newState]) {
-      const newProperty = propertyMap[newState];
+    // handle liquidated -> closed = liquidatedClosed
+    const newStateModified =
+      oldState === VAULT_STATES.LIQUIDATED && newState === VAULT_STATES.CLOSED
+        ? VAULT_STATES.LIQUIDATED_CLOSED
+        : newState;
+
+    if (newStateModified && propertyMap[newStateModified]) {
+      const newProperty = propertyMap[newStateModified];
       (vaultState as any)[newProperty] += BigInt(1);
```

Comment:
I am having a little hard time understanding:
```js
      const newProperty = propertyMap[newStateModified];
      (vaultState as any)[newProperty] += BigInt(1);
```

Here is what I think is happening:

You access the value from propertyMap. I reckon those are vault states(`string`). 
Now after getting the vault state, are we incrementing the string at `(vaultState as any)[newProperty] += BigInt(1);`?

Reply:
we are incrementing the value at `(vaultState as any)[newProperty]` (which is a BigInt) by 1